datasets/scitldr/scitldr.py

```diff
@@ -0,0 +1,184 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Dataset for TLDR: Extreme Summarization of Scientific Documents"""
+
+from __future__ import absolute_import, division, print_function
+
+import json
+import os
+
+import datasets
+
+
+_SOURCE = "source"
+_TARGET = "target"
+
+_CITATION = """\
+@article{cachola2020tldr,
+  title={{TLDR}: Extreme Summarization of Scientific Documents},
+  author={Isabel Cachola and Kyle Lo and Arman Cohan and Daniel S. Weld},
+  journal={arXiv:2004.15011},
+  year={2020},
+}
+"""
+
+_DESCRIPTION = """\
+A new multi-target dataset of 5.4K TLDRs over 3.2K papers.
+SCITLDR contains both author-written and expert-derived TLDRs,
+where the latter are collected using a novel annotation protocol
+that produces high-quality summaries while minimizing annotation burden.
+"""
+
+
+_LICENSE = "Apache License 2.0"
+
+# TODO: Add link to the official dataset URLs here
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URLs = {
+    "Abstract": "https://raw.githubusercontent.com/allenai/scitldr/master/SciTLDR-Data/SciTLDR-A/",
+    "AIC": "https://raw.githubusercontent.com/allenai/scitldr/master/SciTLDR-Data/SciTLDR-AIC/",
+    "FullText": "https://raw.githubusercontent.com/allenai/scitldr/master/SciTLDR-Data/SciTLDR-FullText/",
+}
+
+_TRAIN_DATA = "train.jsonl"
+_TEST_DATA = "test.jsonl"
+_VALID_DATA = "dev.jsonl"
+
+
+# There are several preprocessing scripts given in the original SciTLDR GitHub repository to preprocess this data.
+class SciTldr(datasets.GeneratorBasedBuilder):
```

Comment:
```suggestion
class Scitldr(datasets.GeneratorBasedBuilder):
```
the class name must be the camel case version of the dataset script name `scitldr`